File: jecnaapi-java/src/examples/java/GradesExample.java
import me.tomasan7.jecnaapi.data.grade.Grade;
import me.tomasan7.jecnaapi.data.grade.GradesPage;
import me.tomasan7.jecnaapi.data.grade.Subject;
import me.tomasan7.jecnaapi.data.grade.WeightedGradeAverageCalculator;
import me.tomasan7.jecnaapi.java.JecnaClientJavaWrapper;
import me.tomasan7.jecnaapi.util.SchoolYear;
import me.tomasan7.jecnaapi.util.SchoolYearHalf;
import me.tomasan7.jecnaapi.web.Auth;

import java.util.List;

public class GradesExample
{
    public static void main(String[] args)
    {
        /* Metody join() používáme aby jsme z asynchroního běhu "udělali" synchroní; pouze v příkladech */

        JecnaClientJavaWrapper jecnaClient = new JecnaClientJavaWrapper();

        jecnaClient.login("user", "password").join();

        /* Stáhne známky z roku 2021/2022 z druhého pololetí. */
        GradesPage gradesPage = jecnaClient.getGradesPage(new SchoolYear(2021), SchoolYearHalf.SECOND).join();

        Subject mathSubject = gradesPage.get("Matematika");

        /* Průmer známek */
        float mathAverage = mathSubject.getGrades().average();

        System.out.println("Průměr z matematiky: " + mathAverage);

        /* Do .get() patří string, který určuje část předmětu. (např. Teorie, Cvičení)
         * Pokud předmět není rozdělen, použijte null. */
        List<Grade> mathGrades = mathSubject.getGrades().get(null);

        System.out.println("Známky z matematiky: ");

        for (Grade grade : mathGrades)
            System.out.println(grade);
    }
}

File: src/examples/kotlin/GradesExample.kt
package examples

import kotlinx.coroutines.runBlocking
import me.tomasan7.jecnaapi.JecnaClient
import me.tomasan7.jecnaapi.util.SchoolYear
import me.tomasan7.jecnaapi.util.SchoolYearHalf

fun main(): Unit = runBlocking {

    val client = JecnaClient()

    client.login("user", "password")

    /* Stáhne známky z roku 2021/2022 z druhého pololetí. */
    val gradesPage = client.getGradesPage(SchoolYear(2021), SchoolYearHalf.SECOND)

    val mathSubject = gradesPage["Matematika"]!!

    val mathAverage = mathSubject.grades.average()

    println("Průměr z matematiky: $mathAverage")

    /* Do grades[] patří string, který určuje část předmětu. (např. Teorie, Cvičení)
    * Pokud předmět není rozdělen, použijte null. */
    val mathGrades = mathSubject.grades[null]!!

    println("Známky z matematiky: ")

    for (grade in mathGrades)
        println(grade.valueChar() + " - " + grade.description)
}
